ouroboros-consensus/test-storage/Test/Ouroboros/Storage/ChainDB/Model.hs

```diff
@@ -326,18 +326,15 @@ advanceCurSlot cfg curSlot m =
     advance :: SlotNo -> Model blk -> Model blk
     advance slot m' = m' { currentSlot = slot `max` currentSlot m' }
 
-addBlock :: forall blk. ProtocolLedgerView blk
+addBlock :: forall blk. (ProtocolLedgerView blk, ModelSupportsBlock blk)
          => NodeConfig (BlockProtocol blk)
          -> blk
          -> Model blk -> Model blk
 addBlock cfg blk m
     -- If the block is as old as the tip of the ImmutableDB, i.e. older than
-    -- @k@, we ignore it, as we can never switch to it. TODO what about EBBs?
-  | At (Block.blockNo blk) <= immutableBlockNo secParam m
-    -- Unless we're adding the first block to the empty chain: the empty chain
-    -- has the same block number as the genesis EBB, i.e. 0, so we don't want
-    -- to ignore it in this case.
-  , not addingGenesisEBBToEmptyDB
+    -- @k@, we ignore it, as we can never switch to it.
+  | At (Block.blockNo blk) <= immBlockNo
+  , not (At (Block.blockNo blk) == immBlockNo && isEBB (getHeader blk) == IsEBB)
```

Comment:
So this is basically the same change from the real DB to the model.

Reply:
Indeed.

Reply:
In my last version, I reuse the new function from `.ChainSel` in the model.